pyproject.toml

```diff
@@ -35,7 +35,6 @@ dependencies = [
     "Jinja2>=3.0.0",
     "msgpack",
     "packaging>=22.0",
-    "pydantic>=1.10.0",
```

Comment:
why are we removing this?

Reply:
looks like force push otherwise, I restored it https://github.com/astronomer/astronomer-cosmos/pull/1376#discussion_r1878376966. thanks for pointing out

Reply:
Restored it,